src/coreclr/jit/codegenxarch.cpp

```diff
@@ -6063,6 +6063,14 @@ void CodeGen::genCompareInt(GenTree* treeNode)
     }
     else
     {
+        // "INC REG" instead of "CMP REG, 0xFFFFFFFF" (or just - if "CMP REG, -1")
+        if (op1->isUsedFromReg() && op2->IsIntegralConst(-1) && targetReg != REG_NA)
+        {
+            emit->emitIns_R(INS_inc, emitTypeSize(tree), op1->GetRegNum());
```

Comment:
Not sure I understand this optimization - doesn't it modify op1 reg?

Reply:
Oh, good point! My big mistake. Yes, it changes the logic of the program by modify op1 value. Thank you.